typescript/src/errors.ts

```diff
@@ -11,10 +11,22 @@
 // for each error.
 export class UniffiError extends Error {
   constructor(
-    private readonly __uniffiTypeName: string,
-    private readonly __variantName: string,
-    private readonly __variant: number,
-    message?: string,
+    /*
+     * This member should be private, but typescript requires
+     * it be public because it cannot enforce it.
+     */
+    public readonly __uniffiTypeName: string,
+    /*
+     * This member should be private, but typescript requires
+     * it be public because it cannot enforce it.
+     */
+    public readonly __variantName: string,
+    /*
+     * This member should be private, but typescript requires
+     * it be public because it cannot enforce it.
+     */
+    public readonly __variant: number,
```

Comment:
I wonder if this would be better as private properties in the generated subclasses rather than the super class.

My hunch is that the DX of having `__` properties turn up in autocomplete becomes annoying after a while.

WDYT? Is that annoyance a non-issue, or is it worth the extra work?

Reply:
Yes, that would be better, and they should still be accessible on the union type we're creating. AFAICT it is just the inheritence + const composure that is the problem.

It would be very nice if the __ didn't show up in autocomplete, since you aren't supposed to use them -- makes the generated code a lot more self documenting.